### PR TITLE
[Cody] remove cody-pro feature flag dependency

### PR DESCRIPTION
**Pull request by thenamankumar**
Similar to: https://github.com/sourcegraph/sourcegraph/pull/59660

Remove dependency on cody-pro feature flag which was added during Dec GA release. Things are now stable and it is a good time to cleanup.

## Test plan
- Connect to a local sourcegraph instance
- Visit `/site-admin/feature-flags` and remove the cody-pro feature flag.
- Visit `/cody/manage` and make sure you are on a free plan.
- Check in IDE that Upgrade buttons are showing up.

### CODE REVIEW

File: vscode/src/chat/chat-view/SimpleChatPanelProvider.ts [hunk 1]
```diff
@@ -556,14 +556,7 @@ export class SimpleChatPanelProvider implements vscode.Disposable, ChatSession {
         if (authStatus?.configOverwrites?.chatModel) {
             ChatModelProvider.add(new ChatModelProvider(authStatus.configOverwrites.chatModel))
         }
-        // selection is available to pro only at Dec GA
-        const isCodyProFeatureFlagEnabled = await this.featureFlagProvider.evaluateFeatureFlag(FeatureFlag.CodyPro)
-        const models = ChatModelProvider.get(authStatus.endpoint, this.chatModel.modelID)?.map(model => {
-            return {
-                ...model,
-                codyProOnly: isCodyProFeatureFlagEnabled ? model.codyProOnly : false,
```
Comment (philipp-spiess) [inline vscode/src/chat/chat-view/SimpleChatPanelProvider.ts:564]: Can we remove this flag in the model list too?

**Review by philipp-spiess — COMMENTED** [review]
LGTM but let's also add a case to the test plan when connected to a non dotcom instance (e.g. s2) to make sure we don't bring some of the cody pro functionality to enterprise by mistake

File: vscode/src/chat/chat-view/SimpleChatPanelProvider.ts (hunk 1 above)
Comment (thenamankumar) [reply vscode/src/chat/chat-view/SimpleChatPanelProvider.ts:564]: It was confusing to me too, but here `codyProOnly` means if the model is only available on the pro plan. It has nothing to do with the feature flag but the user's subscription.

Comment (philipp-spiess) [reply vscode/src/chat/chat-view/SimpleChatPanelProvider.ts:564]: Ohh gotcha!

**Review by philipp-spiess — APPROVED** [review]

**Comment by kalanchan** [comment]
thanks @thenamankumar for helping us cleanup!

@abeatrix @DanTup could you help review this?

**Pull request merged**